docs/usage.md

```diff
@@ -0,0 +1,190 @@
+  github:katrinafyi/pac-nix#basil
+```
+
+ddisasm may also be installed through its provided Docker images or
+APT repository (as of Aug 2024, only supports Ubuntu 20.04).
+You can also, of course, build any of these tools manually.
+
+See also: [tool-installation](development/tool-installation.md)
+and [development: building](development/readme.md#building)
+(primarily if you are interested in buliding BASIL manually).
+
+### Preparation
+
+1. Compile a C program into an Aarch64 binary, for example:
+   
+   ```bash
+   aarch64-suse-linux-gcc x.c
+   ```
+
+   You can double-check the produced file with:
+   ```console
+   $ file a.out
+   a.out: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, BuildID[sha1]=ee0d9393526251c23d98f784b86fb3ad694e3517, for GNU/Linux 3.7.0, with debug_info, not stripped
+   ```
+
+2. Disassemble the binary with ddisasm.
+   ```bash
+   ddisasm a.out --ir a.gtirb
+   ```
+
+   This produces a GTIRB protobuffer file.
+   You can inspect its contents, as JSON, with the proto-json.py tool
+   (available in the gtirb-semantics repository and automatically installed by the gtirb-semantics Nix package).
+   ```console
+   $ proto-json.py a.gtirb --seek 8 | jq
+         [...]
+
+         "uuid": "j+onQhA3SSqXO8biANpjIw=="
+       }
+     ],
+     "uuid": "INbyVMRqR9WT3OFbxYALhw==",
+     "version": 4
+   }
+   ```
+
+3. Add instruction semantics to the GTIRB file with gtirb-semantics (.gts means GTIRB with semantics).
+   ```bash
+   gtirb-semantics a.gtirb a.gts
+   ```
+
+   To display this output in a human-readable JSON format, use debug-gts.py
+   (also installed alongside gtirb-semantics).
+   This maps GTIRB basic blocks to source functions and displays instruction semantics
+   alongside the assembly mnemonic.
+   ```console
+   $ debug-gts.py a.gts
+
+   "cwoEFvsGRRm4Myu2pkmWFA==": {
+     "name": "main [entry] [1/1]",
+     "address": 4195972,
+     "code": [
+       {
+         "mov w0, #100                        // =0x64": [
+           "Stmt_Assign(LExpr_Array(LExpr_Var(\"_R\"),0),'0000000000000000000000000000000000000000000000000000000001100100')"
+         ]
+       },
+       {
+         "ret": [
+           "Stmt_Assign(LExpr_Var(\"BTypeNext\"),'00')",
+           "Stmt_Assign(LExpr_Var(\"__BranchTaken\"),Expr_Var(\"TRUE\"))",
+           "Stmt_Assign(LExpr_Var(\"_PC\"),Expr_Array(Expr_Var(\"_R\"),30))"
+         ]
+       }
+     ],
+     "successors": {
+       "RdkcpXXFTX6fJ6AY/CyB+w== / Unresolved ProxyBlock": "Edge.Label(type=Edge.Type.Return, conditional=False, direct=False, )"
+     }
+   },
+   ```
+   
+4. Obtain the symbol table with readelf.
+   ```
+   aarch64-suse-linux-readelf -s -r -W a.out > a.relf
+   ```
+
+### Using BASIL to generate Boogie
+
+This generates a Boogie file in out.bpl.
+```
+basil --input a.gts --relf a.relf -o out.bpl
+```
+(If using BASIL in-repo, substitute `./mill run` in place of `basil`.
```

Comment:
"If using basil compiled from source rather than use the nix package" or similar.

Reply:
I said in repo because it specifically needs the repo checked out to ./mill, but happy with any rewording.

Reply:
I don't think its likely anyone will obtain the source other than by checking out the repo